# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webidl/CookieChangeEvent.webidl
# Commit: 9c8b6e35bf3f
# Full Hash: 9c8b6e35bf3f9916e114ced0473be4b778c7bab1
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-11 09:21:34
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 1 - CookieStore API - WebIDL, r=smaug,edgul
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D215144
# ==============================================================================

diff -r a9b3b52f33a5 -r 9c8b6e35bf3f dom/webidl/CookieChangeEvent.webidl
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/webidl/CookieChangeEvent.webidl	Tue Sep 10 16:31:38 2024 +0000
@@ -0,0 +1,27 @@
+/* -*- Mode: IDL; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * The origin of this IDL file is:
+ * https://wicg.github.io/cookie-store/#CookieChangeEvent
+ */
+
+[Exposed=Window,
+ SecureContext,
+ Pref="dom.cookieStore.enabled"]
+interface CookieChangeEvent : Event {
+  constructor(DOMString type, optional CookieChangeEventInit eventInitDict = {});
+  // TODO: Use FrozenArray once available. (Bug 1236777)
+  // [SameObject] readonly attribute FrozenArray<CookieListItem> changed;
+  // [SameObject] readonly attribute FrozenArray<CookieListItem> deleted;
+  [Frozen, Cached, Constant]
+  readonly attribute sequence<CookieListItem> changed;
+  [Frozen, Cached, Constant]
+  readonly attribute sequence<CookieListItem> deleted;
+};
+
+dictionary CookieChangeEventInit : EventInit {
+  CookieList changed;
+  CookieList deleted;
+};